src/main/java/com/ning/http/util/AsyncHttpProviderUtils.java

```diff
@@ -207,6 +207,19 @@ public static String getNTLM(List<String> authenticateHeaders) {
         return null;
     }
 
+    public static String getHttpHeaderForAuthScheme(List<String> authenticationHeaders, String authScheme)
+    {
+        for (String authenticateHeader : authenticationHeaders)
```

Comment:
Could you optimize this for cases when there are not several headers? You would avoid the `toUpperCase` and the `startsWith`.